insightapi/apimiddleware.go

```diff
@@ -0,0 +1,211 @@
+package insightapi
```

Comment:
as mentioned in comments, please create middlware package since this is a ton of copy/paste from https://github.com/dcrdata/dcrdata/blob/master/apimiddleware.go